test/WebSites/FiltersWebSite/Startup.cs

```diff
@@ -16,10 +16,11 @@ public class Startup
         // Set up application services
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddAuthentication()
+            services.AddAuthentication("Default")
+                    .AddScheme<BasicOptions, EmptyAuthenticationHandler>("Default", _ => { })
                     .AddScheme<BasicOptions, BasicAuthenticationHandler>("Interactive", _ => { })
                     .AddScheme<BasicOptions, BasicAuthenticationHandler>("Api", _ => { });
-            services.AddMvc();
+            services.AddMvc(o => o.CombineAuthorizeFilters = true);
```

Comment:
Well, its a fairly subtle difference in behavior, the existing functional tests had no change with this setting by default, but we do have coverage in the existing unit tests as those failed with the Combine flag set...

I had to add the new logic to the tests that verify the new functionality works (via the counts on the principals).  The most important test for this feature is actually the new `CanReachActionWhenControllerAuthorizeFails` test since that's how many people stumbled into this issue where they have a parent `[Authorize]` (that requires cookies), and a `[Authorize("Bearer"]` on the action, and even when they had a Bearer token, the request would get redirected to the login page since there were two distinct filters being run.

Reply:
I think the point that @kichalla is making is that it's possible to add new startup classes to our existing projects so that you can test both cases if you like, and more importantly without impacting any other tests.

Reply:
We don't really like to see changes to global configuration because it's not clear what the tests expectations were. A case that we were testing might no longer be tested.

Reply:
Ok yeah I can just add a new set of tests that verify this particular new option/behavior in the non superfund site and leave the existing tests alone then :)